### PR TITLE
[backport] Fix decorator usage in tests 

### PR DESCRIPTION
**Pull request by emcastillo**
Backport of #2820

### CODE REVIEW

**Comment by emcastillo** [comment]
[automatic post] Jenkins, test this please.

File: tests/cupy_tests/sorting_tests/test_search.py [hunk 1]
```diff
@@ -217,7 +217,7 @@ class TestNonzero(unittest.TestCase):
     numpy_require = None
 
     @testing.for_all_dtypes()
-    @testing.numpy_cupy_array_list_equal()
+    @testing.numpy_cupy_raises(accept_error=DeprecationWarning)
```
Comment (niboshi) [inline tests/cupy_tests/sorting_tests/test_search.py]: Please undo this.

**Comment by emcastillo** [comment]
PTAL

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Comment by emcastillo** [comment]
Thanks and sorry for my carelessness 🙇

**Comment by niboshi** [comment]
NP!

**Pull request merged**